packages/components/tests/integration/components/hds/code-block/index-test.js

```diff
@@ -0,0 +1,169 @@
+    assert.dom('#test-code-block .token').doesNotExist();
+  });
+
+  // hasCopyButton
+  test('it does not display a Copy button by default', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" />
+    `);
+    assert.dom('.hds-code-block__copy-button').doesNotExist();
+  });
+
+  test(`it displays a Copy button if hasCopyButton is set to true`, async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @hasCopyButton={{true}} />
+    `);
+    assert.dom('.hds-code-block__copy-button').exists();
+  });
+
+  // hasLineNumbers
+  test('it displays line numbers by default', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').hasClass('line-numbers');
+  });
+
+  test('it does not display line numbers if hasLineNumbers is set to false', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @hasLineNumbers={{false}} id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').doesNotHaveClass('line-numbers');
+  });
+
+  // hasLineWrapping
+  test('it does not wrap code line by default', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" id="test-code-block" />
+    `);
+    assert
+      .dom('#test-code-block')
+      .doesNotHaveClass('hds-code-block--has-line-wrapping');
+  });
+
+  test('it wraps code lines if hasLineWrapping is set to true', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @hasLineWrapping={{true}} id="test-code-block" />
+    `);
+    assert
+      .dom('#test-code-block')
+      .hasClass('hds-code-block--has-line-wrapping');
+  });
+
+  // highlightLines
+  test('it highlights the passed in individual line numbers', async function (assert) {
+    await render(hbs`
+    <Hds::CodeBlock
+      id="test-code-block-highlight"
+      @highlightLines={{"1"}}
+      @value="import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from 'dummy/config/environment';
+export default class App extends Application {
+  modulePrefix = config.modulePrefix;
+  podModulePrefix = config.podModulePrefix;
+  Resolver = Resolver;
+}
+loadInitializers(App, config.modulePrefix);"
+    />
+  `);
+    assert
+      .dom('#test-code-block-highlight [data-range="1"]')
+      .exists()
+      .hasClass('line-highlight');
+  });
+
+  // maxHeight
+  test('it uses the passed in maxHeight value', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @maxHeight="100px" />
+    `);
+    assert
+      .dom('.hds-code-block__code')
+      .hasAttribute('style', 'max-height: 100px;');
+  });
+});
```

Comment:
Usually we test the assertions too

Reply:
Indeed, added in [6d132d4](https://github.com/hashicorp/design-system/pull/1687/commits/6d132d4d13c138fac12293ea1b6b0ed947e87d11)